src/Nethermind/Nethermind.JsonRpc/Modules/Trace/TraceModuleFactory.cs

```diff
@@ -31,7 +30,7 @@ public class TraceModuleFactory(
     IPoSSwitcher poSSwitcher,
     ILogManager logManager) : ModuleFactoryBase<ITraceRpcModule>
 {
-    protected readonly IWorldStateManager _worldStateManager = worldStateManager;
+    protected readonly OverridableWorldStateManager _worldStateManager = new(dbProvider, worldStateManager.TrieStore, logManager);
```

Comment:
Shouldn't each TraceModule have their own `OverridableWorldStateManager` instead of all sharing one?
If they all share one overrides from one call, can affect the state of other call?

Reply:
Yes, fixing it.

Used some basic code to reproduce it, but adding unit tests for this case that would be good enough to merge to master is a bit problematic with current implementation - it uses a single `TraceRpcModule` per context.